### ISSUE TITLE
Linux 6.6.1: e1000e doesn't work in sys-net on openQA

### ISSUE THREAD

**Issue by marmarek** [QubesOS/qubes-issues#8709]
[How to file a helpful issue](https://www.qubes-os.org/doc/issue-tracking/)

### Qubes OS release

R4.2

### Brief summary

After updating to [6.6.1](https://github.com/QubesOS/qubes-linux-kernel/pull/856) (both dom0 and sys-net), sys-net can no longer connect to the network when running on openQA (Qubes nested in KVM).

Linux 6.5.8 worked fine.

### Steps to reproduce

Run openQA tests (install Qubes OS inside KVM, set network card to emulated e1000e - not virtio-net, switch sys-net to PV).

### Expected behavior

sys-net works normally

### Actual behavior

```
[2023-11-13 14:07:51] [   60.310138] e1000e 0000:00:00.0 enp0s0: Reset adapter unexpectedly
[2023-11-13 14:07:52] [   60.980714] e1000e 0000:00:00.0 enp0s0: NIC Link is Up 1000 Mbps Full Duplex, Flow Control: Rx/Tx
```
the above repeats every couple of seconds (about 5s), sys-net never gets an IP address.

**Comment by marmarek** [comment]
FWIW, 6.7-rc1 is affected too (in other words - nobody fixed it in the meantime).

**Comment by DemiMarie** [comment]
I’m pretty sure this is just a bug in Xen, Linux (either guest or host), or QEMU.  I don’t think our code has anything to do with it.

Does the emulated e1000e NIC work in:

- A Linux guest?
- A Xen PV dom0?
- A Xen PV guest with nested virt off?
- A Xen HVM guest with nested IOMMU?
- A nested KVM guest with nested IOMMU provided to L1 (but not L2)?

**Comment by marmarek** [comment]
It is clearly regression in Linux, as it was working correctly in Linux 6.5 (see the issue description). I created an issue here, because it affects our CI, and because of that, this issue is a blocker for updating beyond 6.5.x (otherwise we would be left with all testing broken, which is not acceptable).

**Comment by DemiMarie** [comment]
@marmarek is using virtio instead of e1000e an option?

**Comment by marmarek** [comment]
> @marmarek is using virtio instead of e1000e an option?

No, that doesn't work with PCI passthrough into sys-net (Linux gets confused which kind of virtio to use - KVM vs Xen).

**Comment by DemiMarie** [comment]
> > @marmarek is using virtio instead of e1000e an option?
> 
> No, that doesn't work with PCI passthrough into sys-net (Linux gets confused which kind of virtio to use - KVM vs Xen).

So _another_ Linux bug.

**Comment by marmarek** [comment]
`git bisect run` in progress, I guess I'll have result tomorrow-ish

**Comment by marmarek** [comment]
By comparing console logs, the obvious difference is that "bad" case doesn't have logs related to swiotlb. I do see some related changes between 6.5 and 6.6, so that's my current guess.

**Comment by DemiMarie** [comment]
Hopefully the bisection will find an answer.

**Comment by marmarek** [comment]
We'll know soon enough (`1 revisions left to test after this`). It was a bumpy ride, due to some other bugs in the bisection range (that were also fixed soon after).

**Comment by marmarek** [comment]
The bisect said: https://github.com/torvalds/linux/commit/f9a38ea5172a3365f4594335ed5d63e15af2fd18
So, this looks more like configuration mistake on openqa than a kernel bug. e820_host is indeed specifically [disabled on openqa](https://github.com/QubesOS/openqa-tests-qubesos/blob/main/extra-files/qubesteststub/__init__.py#L42-L43), with a comment that it conflicts with booting on OVMF. And I can confirm it indeed (still) makes sys-net crash on OVMF (but with this workaround removed, sys-net and e1000e works on legacy boot).

So, the good part is that I know how to unbreak tests on legacy boot. Now I need to find a good way to unbreak them on OVMF. I have a few ideas, don't need help (yet).

**Comment by marmarek** [comment]
This affected only PV, which isn't relevant anymore.
